### ISSUE TITLE
Modify a widget's settings before rendering

### ISSUE THREAD

**Issue by AmibeWebsites** [pojome/elementor#6101]
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

I am attempting to modify a widget's settings using the elementor/frontend/{$element_type}/before_render hook.

What I expect is that modified settings will be used in the rendering of the widget.

What happens is that some are, and some are not. I'm not 100% sure but it seems like those that work are used in HTML output and those that don't as used in generating CSS.

## Steps to reproduce

In the following code example, for a header widget, the title is changed but the background is not.
```
add_action( 'elementor/frontend/widget/before_render', function( $widget ) {

  $settings = $widget->get_active_settings();

  $widget->set_settings( 'title', 'Heading: ' . $settings[ 'title' ]); // <-- This works
  $widget->set_settings( '_background_image', [ 'url' => 'http://my.dom/img.png', ] ); // <-- This doesn't work

}, 10 , 1);
```
## Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

Can't locate Elementor => System info

**Comment by bainternet** [comment]
@AmibeWebsites 	

Anything used in the generated CSS files are already generated by the time the `elementor/frontend/widget/before_render` action is fired so changing/updating them will have no effect on the already generated CSS file, unless you fulsh the generated files.

thanks.

**Comment by AmibeWebsites** [comment]
@bainternet Thank you, I'll look into the generation of CSS files to understand it a bit better and see how I can work with it.

However, since the update to 2.3.1 the title (in my simplified example) is also not changing and so I've had to rollback updating for now.

Looking into the code of /includes/base/element-base.php these lines (631 - 633):
```
ob_start();
$this->_print_content();
$content = ob_get_clean();
```

...the occur prior to this (line 663):
```
do_action( "elementor/frontend/{$element_type}/before_render", $this );
```

If I move the $this->_print_content() trio to after the do_action I get the results I did previously (so far as the title change is concerned) but I'm not sure if that has other unintended effects or even if that is the desired behaviour? This is how it was in 2.3.0 - not sure why it's been switched around.
